eclipse/src/saros/ui/util/SWTUtils.java

```diff
@@ -221,18 +221,6 @@ public static Shell getShell() {
 
     if (display.isDisposed()) return null;
 
-    final IWorkbench workbench = PlatformUI.getWorkbench();
-
-    final IWorkbenchWindow activeWorkbenchWindow = workbench.getActiveWorkbenchWindow();
-
-    if (activeWorkbenchWindow != null && !activeWorkbenchWindow.getShell().isDisposed())
-      return activeWorkbenchWindow.getShell();
-
-    final IWorkbenchWindow[] workbenchWindows = workbench.getWorkbenchWindows();
-
-    if (workbenchWindows.length > 0 && !workbenchWindows[0].getShell().isDisposed())
-      return workbenchWindows[0].getShell();
-
     return display.getActiveShell();
```

Comment:
Have you read the comment on line 213 (Hint: getActiveShell() may return null if the application is minimized which results in dialogs etc may pop up even if Eclipse is minimized).